homeassistant/components/foscam/number.py

```diff
@@ -0,0 +1,95 @@
+"""Foscam number platform for Home Assistant."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from libpyfoscamcgi import FoscamCamera
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import FoscamConfigEntry, FoscamCoordinator
+from .entity import FoscamEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class FoscamNumberEntityDescription(NumberEntityDescription):
+    """A custom entity description with adjustable features."""
+
+    native_value_fn: Callable[..., int]
+    set_value_fn: Callable[[FoscamCamera, float], Any]
+    exists_fn: Callable[..., int]
+
+
+NUMBER_DESCRIPTIONS: list[FoscamNumberEntityDescription] = [
+    FoscamNumberEntityDescription(
+        key="device_volume",
+        translation_key="device_volume_entity",
+        native_min_value=0,
+        native_max_value=100,
+        native_step=1,
+        native_value_fn=lambda data: data.device_volume,
+        set_value_fn=lambda session, value: session.setAudioVolume(value),
+        exists_fn=lambda coordinator: 1,
+    ),
+    FoscamNumberEntityDescription(
+        key="speak_volume",
+        translation_key="speak_volume_entity",
+        native_min_value=0,
+        native_max_value=100,
+        native_step=1,
+        native_value_fn=lambda data: data.speak_volume,
+        set_value_fn=lambda session, value: session.setSpeakVolume(value),
+        exists_fn=lambda coordinator: coordinator.data.supports_speak_volume_adjustment,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: FoscamConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up foscam number from a config entry."""
+    coordinator = config_entry.runtime_data
+    async_add_entities(
+        [
+            FoscamVolumeNumberEntity(coordinator, description)
+            for description in NUMBER_DESCRIPTIONS
+            if description.exists_fn is None or description.exists_fn(coordinator)
+        ]
+    )
```

Comment:
```suggestion
    async_add_entities(
        FoscamVolumeNumberEntity(coordinator, description)
        for description in NUMBER_DESCRIPTIONS
        if description.exists_fn(coordinator)
    )
```
The function isn't nullable